lib/facade_converter.ts

```diff
@@ -147,6 +147,48 @@ export class FacadeConverter extends base.TranspilerBase {
     this.emit(ident);
   }
 
+  shouldEmitNew(c: ts.CallExpression): boolean {
+    if (!this.tc) return true;
+
+    var symbol: ts.Symbol;
+    var ident: string;
+
+    if (c.expression.kind === ts.SyntaxKind.Identifier) {
+      // Function call.
+      ident = base.ident(c.expression);
+
+      if (!this.candidateProperties.hasOwnProperty(ident)) return true;
+      symbol = this.tc.getSymbolAtLocation(c.expression);
+      if (FACADE_DEBUG) console.log('s:', symbol);
+
+      if (!symbol) {
+        this.reportMissingType(c, ident);
+        return true;
+      }
+    } else if (c.expression.kind === ts.SyntaxKind.PropertyAccessExpression) {
+      // Method call.
+      var pa = <ts.PropertyAccessExpression>c.expression;
```

Comment:
Can you de-duplicate this code some more? I think if you create a variable for `ident` and `symbolLocation` or so, you can share most of the code.